pkg/k8shandler/secret_test.go

```diff
@@ -0,0 +1,265 @@
+
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	rbac "k8s.io/api/rbac/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestClusterLoggingRequest_CreateOrUpdateSecretNoop(t *testing.T) {
+	clusterRoleBinding := &rbac.ClusterRoleBinding{}
+
+	clusterLoggingRequest := &ClusterLoggingRequest{
+		client:  fake.NewFakeClient(clusterRoleBinding),
+		cluster: &logging.ClusterLogging{},
+	}
+
+	// mimic ca.crt
+	caCrtStr := `-----BEGIN CERTIFICATE-----
+This is a ca cert.
+-----END CERTIFICATE-----`
+
+	// mimic ca.key
+	caKeyStr := `-----BEGIN PRIVATE KEY-----
+This is a private key.
+-----END PRIVATE KEY-----`
+
+	initSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
+
+	// create
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(initSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 1 [%v]", err)
+	}
+
+	firstSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 1 [%v]", err)
+	}
+
+	// update
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(firstSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 2 [%v]", err)
+	}
+
+	secondSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 2 [%v]", err)
+	}
+
+	if reflect.DeepEqual(firstSecret, secondSecret) {
+		t.Logf("Initial secret [%p]\n%v", initSecret, initSecret)
+		t.Logf("First secret [%p]\n%v", firstSecret, firstSecret)
+		t.Logf("Second secret [%p]\n%v", secondSecret, secondSecret)
+	} else {
+		t.Errorf("First secret [%v] != Second secret [%v]", firstSecret, secondSecret)
+	}
+}
+
+func TestClusterLoggingRequest_CreateOrUpdateSecretSameKeyNewCert(t *testing.T) {
+	clusterRoleBinding := &rbac.ClusterRoleBinding{}
+
+	clusterLoggingRequest := &ClusterLoggingRequest{
+		client:  fake.NewFakeClient(clusterRoleBinding),
+		cluster: &logging.ClusterLogging{},
+	}
+
+	// mimic ca.crt
+	caCrtStr := `-----BEGIN CERTIFICATE-----
+This is a ca cert.
+-----END CERTIFICATE-----`
+
+	// mimic ca.key
+	caKeyStr := `-----BEGIN PRIVATE KEY-----
+This is a private key.
+-----END PRIVATE KEY-----`
+
+	initSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
+			"testkey": []byte(caKeyStr),
+		})
+
+	// create
+	if err := clusterLoggingRequest.CreateOrUpdateSecret(initSecret); err != nil {
+		t.Errorf("CreateOrUpdateSecret failed 1 [%v]", err)
+	}
+
+	firstSecret, err := clusterLoggingRequest.GetSecret("test-secret")
+	if err != nil {
+		t.Errorf("GetSecret failed 1 [%v]", err)
+	}
+
+	caCrtStr = `-----BEGIN CERTIFICATE-----
+This is a new ca cert.
+-----END CERTIFICATE-----`
+
+	newSecret := NewSecret(
+		"test-secret",
+		clusterLoggingRequest.cluster.Namespace,
+		map[string][]byte{
+			"testca":  []byte(caCrtStr),
```

Comment:
This is the `TestClusterLoggingRequest_CreateOrUpdateSecretSameKeyNewCert` - the key `testca` is the same, but it has a new value `This is a new ca cert`.  Same key, new cert value.

Reply:
Ah, ok.  I was just changing the values, not the key of the secret maps...  Fixing them.

But why the ci/prow/unit test failed?  I cannot reproduce the failure on my laptop...

$ go test -v | egrep TestClusterLoggingRequest
=== RUN   TestClusterLoggingRequest_CreateOrUpdateSecretNoop
--- PASS: TestClusterLoggingRequest_CreateOrUpdateSecretNoop (0.00s)
=== RUN   TestClusterLoggingRequest_CreateOrUpdateSecretSameKeyNewCert
--- PASS: TestClusterLoggingRequest_CreateOrUpdateSecretSameKeyNewCert (0.00s)
=== RUN   TestClusterLoggingRequest_CreateOrUpdateSecretNewKeySameCert
--- PASS: TestClusterLoggingRequest_CreateOrUpdateSecretNewKeySameCert (0.00s)
=== RUN   TestClusterLoggingRequest_CreateOrUpdateSecretRemovingKey
--- PASS: TestClusterLoggingRequest_CreateOrUpdateSecretRemovingKey (0.00s)